api/app/controllers/api/v1/positions_controller.rb

```diff
@@ -1,92 +1,92 @@
+                render json: { status: 'error', message: position.errors, payload: {} }
+            end
+            params[:position_id] = position[:id]
+            message = valid_ad_and_matching
+            if not message
+                render json: { status: 'success', message: '', payload: position }
+            else
+                render json: { status: 'success', message: message, payload: {} }
+            end
         end
-      end
-    end
 
-    # PATCH/PUT /positions/1
-    def update
-      if @position.update(position_params)
-        render json: @position
-      else
-        render json: @position.errors, status: :unprocessable_entity
-      end
-    end
+        private
+        # Only allow a trusted parameter "white position" through.
+        def position_params
+            params.permit(
+                :session_id,
+                :position_code, 
+                :position_title, 
+                :est_hours_per_assignment, 
+                :est_start_date, 
+                :est_end_date, 
+                :position_type, 
+            )
+        end
 
-    # DELETE /positions/1
-    def destroy
-      if @position.destroy
-        head :no_content, status: :ok
-      else
-        render json: @position.errors, status: :unprocessable_entity
-      end
-    end
+        def position_data_for_ad_params
+            params.permit(
+                :position_id,
+                :duties, 
+                :qualifications, 
+                :ad_hours_per_assignment, 
+                :ad_num_assignments, 
+                :ad_open_date, 
+                :ad_close_date, 
+            )
+        end
 
-    private
+        def position_data_for_matching_params
+            params.permit(
+                :position_id,
+                :desired_num_assignments, 
+                :current_enrollment, 
+                :current_waitlisted
+            )
+        end
 
-    # Use callbacks to share common setup or constraints between actions.
-    def set_position
-      @position = Position.find(params[:id])
-    end
+        def positions_by_session
+            return Position.order(:id).select do |entry|
+                entry[:session_id] == params[:session_id].to_i
+            end
+        end
 
-    # Only allow a trusted parameter "white position" through.
-    def position_params
-      params.permit(
-        :cap_enrolment, :course_code, :course_name, :current_enrolment, :duties, :hours,
-        :num_waitlisted, :openings, :qualifications, :session_id,
-        :start_date, :end_date
-      )
+        def valid_ad_and_matching
+            ad = PositionDataForAd.new(position_data_for_ad_params)
+            matching = PositionDataForMatching.new(position_data_for_matching_params)
+            if ad.save and matching.save
+                return nil
+            elsif ad.save and not matching.save
+                return matching.errors
+            elsif not ad.save and matching.save
```

Comment:
Does this code re-execute `*.save` a bunch of times? Will it save items to the database multiple times??

Reply:
I don't know if this re-execute a bunch of times, so I'll just save the outcome to a local variable.